docs/html/user_guide.rst

```diff
@@ -79,6 +79,21 @@ as the "username" and do not provide a password, for example -
 ``https://0123456789abcdef@pypi.company.com``
 
 
+Keyring Support
+---------------
+
+pip also supports credentials stored in your keyring using the `keyring`_
+library.
```

Comment:
Maybe add an explicit note that `keyring` is not supplied as part of pip, and that you need to install it yourself so that pip can access it?